File: BookStore-BE/src/main/java/com/java/bookstore/entities/OrderItemEntity.java
package com.java.bookstore.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.Min;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "order_items")
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class OrderItemEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Min(value = 0)
	private int quantity;

	private double totalPrice;
	
	private double currentPrice;
	
	@ManyToOne
	@JoinColumn(name = "book_id")
	private BookEntity book;
	
	@ManyToOne
	@JoinColumn(name = "order_id")
	private OrderEntity order;
}

File: BookStore-NestJS/src/entities/order-item/entities/order-item.entity.ts
import { Book } from "src/entities/books/entities/book.entity";
import { Order } from "src/entities/orders/entities/order.entity";
import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from "typeorm";

@Entity()
export class OrderItem {
    @PrimaryGeneratedColumn()
    id: number;

    @Column('double', { precision: 15, scale: 8 })
    currentPrice: number;

    @ManyToOne(() => Order, order => order.orderItems)
    order: Order;

    @ManyToOne(() => Book)
    book: Book;

    @Column()
    quantity: number;

    @Column('float')
    totalPrice: number;
}
